server/controllers/departuresController.ts

```diff
@@ -19,30 +19,31 @@ export default class DeparturesController {
       const { premisesId, bookingId } = req.params
       const booking = await this.bookingService.getBooking(premisesId, bookingId)
       const premisesSelectList = await this.premisesService.getPremisesSelectList()
+      const referenceData = await this.departureService.getReferenceData()
 
-      res.render('departures/new', { premisesId, booking, premisesSelectList })
+      res.render('departures/new', { premisesId, booking, premisesSelectList, referenceData })
     }
   }
 
   create(): RequestHandler {
     return async (req: Request, res: Response) => {
       const { premisesId, bookingId } = req.params
-      const body = req.body as DepartureDto
-      const { dateTime } = convertDateAndTimeInputsToIsoString(body, 'dateTime')
+      const { dateTime } = convertDateAndTimeInputsToIsoString(req.body, 'dateTime')
 
-      const departure: Omit<Departure, 'id' | 'bookingId'> = {
-        ...body.departure,
+      const departure = {
+        ...req.body.departure,
         dateTime,
-      }
+      } as DepartureDto
 
       try {
         const { id } = await this.departureService.createDeparture(premisesId, bookingId, departure)
         res.redirect(`/premises/${premisesId}/bookings/${bookingId}/departures/${id}/confirmation`)
       } catch (err) {
         const booking = await this.bookingService.getBooking(premisesId, bookingId)
         const premisesSelectList = await this.premisesService.getPremisesSelectList()
+        const referenceData = await this.departureService.getReferenceData()
```

Comment:
Is there not a chance that any of these 3 `await` statements could fail in the same way as the one in the `try` block?

Reply:
We tend to use store errors and input in flash and redirect back to the GET - that has the benefit of meaning you don't have to maintain all the code required to render a page in two places.

Reply:
maybe `renderWithErrors` would be better named `renderAnyValidationErrorsOrPropagate` or something - something to hint it will only render errors it knows what to do with

Reply:
I think with this pattern, it means if you get a one off error or timeout, to render the standard error page, it would have to do these 3 calls, wait for a response, throw the responses away and then let the standard error handler do its thing.

Reply:
Oh that's interesting. I'll give that a try. Mind if I do that in a seperate PR to stop this getting bloated? That way I can change the pattern everywhere

Reply:
What happens with the user's previous answers? Do you store the body in the flash too?

Reply:
Ah, have seen what happens elsewhere. Yep. Agreed, this is a much better pattern